force_gromacs/pipeline/gromacs_pipeline.py

```diff
@@ -1,38 +1,51 @@
 from itertools import islice
 
 from traits.api import (
-    List, Tuple, Unicode, on_trait_change, Dict
+    List, Tuple, Unicode, on_trait_change, Dict, Property
 )
 
 from force_gromacs.core.base_gromacs_process import BaseGromacsProcess
 
 
 class GromacsPipeline(BaseGromacsProcess):
-    """ A simple pipeline for Gromacs commands, based on scikit-learn
+    """A simple pipeline for Gromacs commands, based on scikit-learn
     pipeline functionality that can sequentially apply a list of Gromacs
     commands using subprocess and retain the standard output/error."""
 
+    # --------------------
+    #  Regular Attributes
+    # --------------------
+
     #: List of tuples (name, BaseGromacsProcess) objects that are chained,
     #: in the order, in which they are chained.
     steps = List(Tuple(Unicode, BaseGromacsProcess))
 
     #: Output from the most recent Gromacs run
     run_output = Dict()
 
+    # --------------------
+    #      Properties
+    # --------------------
+
+    #: A dictionary representing `steps` with keys as the first element
+    #: and values as the second element in each tuple
+    named_steps = Property(Dict, depends_on='steps[]')
+
+    # --------------------
+    #      Listeners
+    # --------------------
+
+    def _get_named_steps(self):
+        return dict(**dict(self.steps))
```

Comment:
`dict(self.steps)` also works for the tests I have. Can you please confirm that?

Reply:
Yep, that works - still can't remember why I didn't remove that weird syntax. Thanks for pointing this out